content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm

```diff
@@ -107,9 +107,7 @@ includeLatestJQuery('sakai_filepicker_attach.vm');
 <script src="/library/js/spinner.js" type="text/javascript"></script>
 
 <div class="portletBody specialLink" id="portletBody">
-	#if ($alertMessage)
```

Comment:
It seems like if you remove this if check completely that it would always display the alert banner. Is this not the case? I'd feel the warning should only be there if there is a message to display.

Reply:
Note that the alertMessage div has a bootstrap 'd-none' class so it will not display until the client side javascript (from SAK-47869) toggles that off if it detect that the attempted file has a size exceeding the maximum.
Thanks for catching and filing SAK-50147 by the way :)

Reply:
Okay great, I didn't see that was the pattern here now. 

It _looks_ like it's not removed the default
`		if (!needToCheckFileSize) { `

only in the else case? Were you seeing it removed? 

I noticed this as part of the Workflow on https://sakaiproject.atlassian.net/browse/SAK-50113

Thanks!

Reply:
@jonespm   It has been a while since I looked at my code. I believe the file size is only checked (client-side) at the time a file is picked. If it exceeds the size limit then the message banner class is changed from d-none to d-inline-block, making it visible.

If the user changes tabs, the message banner is changed back to d-none to hide it.
Other cases in the work flow:

 1. If a new file is selected then its size is checked and if it is over the limit, the banner will remain but have it wording (filename and size) updated to reflect the new oversized file.
 2. If a new file is selected then its size is checked and is under the limit, the client-side picker form is submitted and upon re-rendering the modal view the alertMessage element once again has a d-none class.
 3. If user select Continue while alert is visible (Continue will only be active is other within limit files have been chosen during this modal 'session') then the client-side form is submitted and the valid attachments are added to the tool's item and the modal is closed.
 4. Closing the modal via Cancel or X do not need to address the message banner since the modal is being closed.